### ISSUE TITLE
As a user, I'd like to see the URLs generated by `pathauto` reflect the page title correctly.

### ISSUE THREAD

**Issue by quamsta** [uiowa/uiowa#5853]
# Story
As a user, I'd like to see the URLs generated by `pathauto` reflect the page title correctly.

# Reasoning
According to [this documentation page on Drupal's site](https://www.drupal.org/node/847370):

> Pathauto has a feature to remove less important words from rewritten URLs in order to save space without compromising semantical meaning, or potentially, usefulness in search engine optimization (SEO).

[A person in the comments section of an issue](https://www.drupal.org/project/pathauto/issues/3298897) found [the original source of this list](https://www.drupal.org/project/pathauto/issues/84553), which dates back to **September 2006** and makes me question its relevance to modern SEO practices and general best practices.

I've had to constantly re-add words to URLs manually (and consequently this removes the benefits of something like `pathauto` automatically generating the URL) when the semantical meaning of the URL has actually been changed by the removal of these words. I don't really understand why these words need to be removed when you can end up with strange URLs that need to be promoted.

A few examples I've found of this going wrong:

- On the Mental Health at Iowa website, I've had to adjust URLs for various posts including "Social Media: What's the Right Fit For You" to put words back in them. For example: `social-media-whats-right-fit-you` had to be manually adjusted to `social-media-whats-the-right-fit-for-you`
- [Someone trying to have a URL containing `internet-of-things` with `of` being stripped out](https://www.drupal.org/project/pathauto/issues/3061797)
- [Someone trying to have a URL containing `talk-with-your-child-about-their-illness` with `with` being stripped out, making it look strange](https://stackoverflow.com/questions/23136495/drupal-automatic-url-alias-removing-words)
- [Someone questioning why `before` is on this list when a word like that could change the semantic definition of the URL, e.g. `what-to-do-before-climbing-a-mountain`](https://www.drupal.org/project/pathauto/issues/3298897).

# Proposed solution

Remove the items in the `ignore_words` setting:
`a, an, as, at, before, but, by, for, from, is, in, into, like, of, off, on, onto, per, since, than, the, this, that, to, up, via, with`

**Comment by quamsta** [comment]
@joewhitsitt said in the draft PR:

> Not disagreeing with you on this, but I'd like to highlight another issue to this before we dump these. As there might be things we are missing.
> 
> I don't really understand why these words need to be removed
> 
> There is a max-length issue as play here as well. I haven't had the time to dig into this further, but if this started as an issue instead of a PR maybe we could spend some time on it. I think it originally came up with Education and their nested Areas of Study.
> 
> Take note of the following pages. Part of the reason to ignore words is to keep the URLs short which becomes more important as pages are nested. Here only the first three letters are used.
> 
> https://sandbox.prod.drupal.uiowa.edu/child-page/test-page/ut-tincidunt-tincidunt-erat-fusce-commodo-aliquam-arcu-phasellus-accumsan/nam
> 
> https://education.uiowa.edu/areas-study/higher-education-and-leadership/higher-education-and-student-affairs/ma-higher
> 
>

**Comment by quamsta** [comment]
@joewhitsitt And it looks like the reason for these truncated URLs might be a `path_alias` table size?

https://www.drupal.org/project/pathauto/issues/2858096

It just seems odd that small words like these have to be truncated due to a technical limitation. But if it's an upstream Drupal issue I can probably rest my case here.

**Comment by joewhitsitt** [comment]
I think that given the technical limitation, the ignore words help prioritize what words should be kept. Obviously, Drupal isn't very smart and this is happening even when there isn't a need to limit, cut it down.

Thank you for pulling together these links for others to review and slowing this down a bit. I'll be honest, I was a bit bothered when this came out out of nowhere, mid-sprint, a platform-wide change and that you requested approval to merge.

As a user, I value human friendly URLs so I think this change is fine. The ability to override is still there for the situations where it comes up short. Something to keep in mind is that on the next save, content will be updated and an automatic redirect will be created (similar if they changed the content title). Currently redirects do not expire from the table.

**Comment by quamsta** [comment]
@joewhitsitt Yeah, sorry for breezing in out of nowhere with the PR. I realized that was a bit much and decided to slow my roll a bit. I thought I had left the original PR as a draft to at least facilitate discussion but I think I marked it for review and requested review out of habit without much thought.

> Something to keep in mind is that on the next save, content will be updated and an automatic redirect will be created (similar if they changed the content title). Currently redirects do not expire from the table.

This.. is exactly why I probably needed to slow down a bit. Since this would change a lot of URLs system-wide when any page is changed from here-on-out, I'm a bit worried those redirect tables may fill up quite a bit and a lot of already shortened/word-removed links may turn into redirects.  I'm guessing there's probably no way to disable `pathauto` for older nodes without manually setting their paths?

**Comment by joewhitsitt** [comment]
>I'm a bit worried those redirect tables may fill up quite a bit and a lot of already shortened/word-removed links may turn into redirects.

To ease concerns a bit, of the sites on uiowa.prod, all of the redirect tables are under 10 MB.

Since it only happens on save, I don't think there will be a huge jump.
